File: examples/php/whatsapp_checker.php
<?php
// WhatsApp Number Checker API — bulk verification example (PHP + cURL).
// Workflow: submit a file of E.164 numbers -> poll status -> download results.
// Docs: https://docs.checknumber.ai/whatsapp-bulk-checker

const BASE_URL = "https://api.checknumber.ai";
const TASK_TYPE = "ws"; // ws | ws_active | ws_avatar
$apiKey = getenv("CHECKNUMBER_API_KEY") ?: "YOUR_API_KEY";

function apiPost(string $url, array $fields, string $apiKey): array {
    $ch = curl_init($url);
    curl_setopt_array($ch, [
        CURLOPT_RETURNTRANSFER => true,
        CURLOPT_POST => true,
        CURLOPT_HTTPHEADER => ["X-API-Key: {$apiKey}"],
        CURLOPT_POSTFIELDS => $fields,
    ]);
    $resp = curl_exec($ch);
    $code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);
    if ($code !== 200) {
        throw new Exception("HTTP {$code}: {$resp}");
    }
    return json_decode($resp, true);
}

$submit = apiPost(BASE_URL . "/v1/tasks", [
    "file" => new CURLFile("numbers.txt", "text/plain", "numbers.txt"),
    "task_type" => TASK_TYPE,
], $apiKey);
$taskId = $submit["task_id"];
echo "task_id: {$taskId}\n";

do {
    $task = apiPost(BASE_URL . "/v1/gettasks", ["task_id" => $taskId], $apiKey);
    echo "status={$task['status']} success={$task['success']}/{$task['total']}\n";
    if ($task["status"] === "failed") {
        throw new Exception("task failed");
    }
    if ($task["status"] !== "exported") {
        sleep(5);
    }
} while ($task["status"] !== "exported");

if (!empty($task["result_url"])) {
    file_put_contents("results.xlsx", fopen($task["result_url"], "r"));
    echo "saved to: results.xlsx\n";
}

File: examples/python/whatsapp_checker.py
#!/usr/bin/env python3
"""WhatsApp Number Checker API — bulk verification example.

Workflow: submit a file of E.164 numbers -> poll status -> download results.
Docs: https://docs.checknumber.ai/whatsapp-bulk-checker
"""
import os
import time
import requests

BASE_URL = os.environ.get("API_BASE_URL", "https://api.checknumber.ai")
API_KEY = os.environ.get("CHECKNUMBER_API_KEY", "YOUR_API_KEY")
TASK_TYPE = "ws"  # ws | ws_active | ws_avatar


def submit_task(file_path: str) -> dict:
    with open(file_path, "rb") as f:
        resp = requests.post(
            f"{BASE_URL}/v1/tasks",
            headers={"X-API-Key": API_KEY},
            files={"file": (os.path.basename(file_path), f, "text/plain")},
            data={"task_type": TASK_TYPE},
            timeout=30,
        )
    resp.raise_for_status()
    return resp.json()


def get_task(task_id: str) -> dict:
    resp = requests.post(
        f"{BASE_URL}/v1/gettasks",
        headers={"X-API-Key": API_KEY},
        data={"task_id": task_id},
        timeout=30,
    )
    resp.raise_for_status()
    return resp.json()


def poll(task_id: str, interval: int = 5) -> dict:
    while True:
        task = get_task(task_id)
        print(f"status={task['status']} success={task['success']}/{task['total']}")
        if task["status"] == "exported":
            return task
        if task["status"] == "failed":
            raise RuntimeError("task failed")
        time.sleep(interval)


def download(result_url: str, out_path: str = "results.xlsx") -> str:
    resp = requests.get(result_url, stream=True, timeout=300)
    resp.raise_for_status()
    with open(out_path, "wb") as f:
        for chunk in resp.iter_content(chunk_size=8192):
            f.write(chunk)
    return out_path


if __name__ == "__main__":
    task = submit_task("numbers.txt")
    print("task_id:", task["task_id"])
    final = poll(task["task_id"])
    if final.get("result_url"):
        print("saved to:", download(final["result_url"]))
